src/events/logs.ts

```diff
@@ -0,0 +1,385 @@
+        .setColor(PURPLE)
+        .setAuthor({
+            name: "Server profile updated",
+            iconURL: before.guild.iconURL()
+        })
+        .setThumbnail(before.displayAvatarURL())
+        .addFields(
+            { name: "ID", value: before.id, inline: false }
+        );
+
+    // Add this if the nickname was updated
+    if (oldNick !== newNick) {
+        embed.addFields(
+            { name: "Old nickname", value: oldNick ?? before.displayName, inline: false },
+            { name: "New nickname", value: newPfp ?? after.displayName, inline: false },
+            { name: "_ _", value: "_ _", inline: true } // Empty column
+        );
+    }
+
+    // Add this if the avatar was updated
+    if (oldPfp !== newPfp) {
+        embed.addFields(
+            { name: "New avatar", value: "_ _" } // Empty field value
+        )
+            .setImage(after.displayAvatarURL());
+    }
+
+    return embed;
+};
+
+/**
+ * Generates an embed for whenever a member's nickname or server avatar is updated.
+ * 
+ * @param before the user's state before the update
+ * @param after the user's state after the update
+ * @returns an embed for logging; or null if the user's username, display name, and avatar stay the same
+ */
+export const genUserUpdateEmbed = (before: User, after: User): EmbedBuilder | null => {
+    // Properties to check
+    const oldUsername: string = before.username;
+    const newUsername: string = after.username;
+    const oldDisplayName: string = before.globalName;
+    const newDisplayName: string = after.globalName;
+    const oldPfp: string = before.avatar;
+    const newPfp: string = after.avatar;
+
+    // No embed generated if username, display name, and avatar stay the same
+    if (
+        oldUsername === newUsername &&
+        oldDisplayName === newDisplayName &&
+        oldPfp === newPfp
+    ) {
+        return null;
+    }
+
+    const embed: EmbedBuilder = new EmbedBuilder()
+        .setTitle(oldUsername)
+        .setColor(PURPLE)
+        .setThumbnail(before.displayAvatarURL())
+        .addFields(
+            { name: "ID", value: before.id }
+        );
+
+    // Add this if the username was updated
+    if (oldUsername !== newUsername) {
+        embed.addFields(
+            { name: "Old username", value: oldUsername, inline: false },
+            { name: "New username", value: newUsername, inline: false },
+        );
+    }
+
+    // Add this if the display name was updated
+    if (oldDisplayName !== newDisplayName) {
+        embed.addFields(
+            { name: "Old display name", value: oldDisplayName, inline: false },
+            { name: "New display name", value: newDisplayName, inline: false },
+        );
+    }
+
+    // Add this if the username was updated
+    if (oldPfp !== newPfp) {
+        embed.addFields(
+            { name: "New avatar", value: "_ _", inline: false }
+        )
+            .setImage(after.displayAvatarURL());
+    }
+
+    return embed;
+};
```

Comment:
This is the embed used for the listener above
It checks for changes to usernames, display names, and avatars